cmd/zoekt-sourcegraph-indexserver/backoff_test.go

```diff
@@ -0,0 +1,249 @@
+package main
+
+import (
+	"github.com/sourcegraph/log/logtest"
+	"testing"
+	"time"
+)
+
+func TestQueue_BackoffOnFail(t *testing.T) {
+	backoffDuration := 1 * time.Millisecond
```

Comment:
These new tests rely on sleeping to simulate actions during the backoff duration and after backoff. 

We could have `SetIndexed()` accept a time which could remove this need to sleep in our tests but that parameter wouldn't be used outside of the tests (since we aren't concerned with calling time.now() downstream in the queue or backoff code vs. calling it upstream in the server functions).

Reply:
FYI relying on sleep unless used very carefully often leads to flakey tests on CI where the system can be very busy (our buildkite) / under provisioned (like on github actions). I'll comment on each test to try and ensure it isn't flakey: backoffDuration/etc needs to be higher since 1ms may pass due to busy machine. It is fine to do this since we are just checking it isn't allowed to push and we are not sleeping.

Reply:
Yeah, I agree that the sleep has drawbacks in that it'll vary across environments and especially if failures pop up in CI. I'll reply to the other comment you posted about perhaps unit testing just the backoff